dunereco/FDSensOpt/NuAngularReco_module.cc

```diff
@@ -0,0 +1,204 @@
+////////////////////////////////////////////////////////////////////////
+//
+// \file NuAngularReco_module.cc
+//
+// Generated at Mon Oct 30 08:54:19 2023 by Henrique Souza using cetskelgen
+//
+////////////////////////////////////////////////////////////////////////
+
+#ifndef NuAngularReco_H
+#define NuAngularReco_H
+
+//ART
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+//LArSoft
+#include "larcore/Geometry/Geometry.h"
+#include "lardataobj/RecoBase/Wire.h"
+#include "lardataobj/RecoBase/Track.h"
+#include "lardataobj/RecoBase/Hit.h"
+#include "lardataobj/RecoBase/Shower.h"
+#include "lardata/Utilities/AssociationUtil.h"
+#include "lardata/DetectorInfoServices/DetectorClocksService.h"
+#include "lardata/DetectorInfoServices/DetectorPropertiesService.h"
+#include "larpandora/LArPandoraInterface/LArPandoraHelper.h"
+//DUNE
+#include "dunereco/FDSensOpt/FDSensOptData/AngularRecoOutput.h"
+#include "dunereco/FDSensOpt/NeutrinoAngularRecoAlg/NeutrinoAngularRecoAlg.h"
+#include "dunereco/AnaUtils/DUNEAnaEventUtils.h"
+#include "dunereco/AnaUtils/DUNEAnaHitUtils.h"
+#include "dunereco/AnaUtils/DUNEAnaShowerUtils.h"
+
+#include <memory>
+
+namespace dune {
+
+
+  class NuAngularReco : public art::EDProducer {
+    public:
+      explicit NuAngularReco(fhicl::ParameterSet const& p);
+      // The compiler-generated destructor is fine for non-base
+      // classes without bare pointers or other resource use.
+
+      // Plugins should not be copied or assigned.
+      NuAngularReco(NuAngularReco const&) = delete;
+      NuAngularReco(NuAngularReco&&) = delete;
+      NuAngularReco& operator=(NuAngularReco const&) = delete;
+      NuAngularReco& operator=(NuAngularReco&&) = delete;
+
+      // Required functions.
+      void produce(art::Event& evt) override;
+
+      // Selected optional functions.
+      void beginJob() override;
+      void endJob() override;
+
+    private:
+            art::Ptr<recob::Track> GetLongestTrack(const art::Event& event);
+            art::Ptr<recob::Shower> GetHighestChargeShower(detinfo::DetectorClocksData const& clockData,
```

Comment:
Not something to address now, but we should think about moving these functions outside of here (and out of the neutrino energy alg) so that there's a single point of maintenance